lib/digital_signal/digital_signal.h

```diff
@@ -5,23 +5,53 @@
 #include <stdbool.h>
 
 #include <furi_hal_gpio.h>
+#include <stm32wbxx_ll_dma.h>
+#include <stm32wbxx_ll_tim.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+/* helper for easier signal generation */
+#define DIGITAL_SIGNAL_MS(x) (x * 100000000UL)
+#define DIGITAL_SIGNAL_US(x) (x * 100000UL)
+#define DIGITAL_SIGNAL_NS(x) (x * 100UL)
+#define DIGITAL_SIGNAL_PS(x) (x / 10UL)
+
 typedef struct {
```

Comment:
Does it has to be public?

Reply:
these were meant as helper macros to provide users an easy way to call code like:
`sig->edge_timings[0] = DIGITAL_SIGNAL_US(100);`
it just helps with unit conversion, not requiring calculation, but it's not required and i can remove it if you want.

Reply:
I meant PulseReaderUnit itself. Basically this structure exposes its internals which makes difficult future porting.

Couple examples how we usually do that:

Public API provided in header file contains anonymous structure and provides getters/setters/methods to work with it: https://github.com/flipperdevices/flipperzero-firmware/blob/dev/applications/services/gui/view.h#L34
Everything else is in implementation or in private API.

Reply:
absolutely. will change it.